### PR TITLE
Document template.select platform

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
This PR and https://github.com/home-assistant/home-assistant.io/pull/18975 will create merge conflicts for each other depending on which one gets merged first.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/54835
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by SNoof85 — COMMENTED** [review]
Wrong copy pastes with number ? :)

File: source/_integrations/template.markdown [hunk 1]
```diff
@@ -168,16 +162,50 @@ binary_sensor:
...
-      required: false
-      type: template
-      default: true
     device_class:
       description: Sets the class of the device, changing the device state and icon that is displayed on the UI (see below). It does not set the `unit_of_measurement`.
       required: false
       type: device_class
       default: None
+number:
+  description: List of numbers
```
Comment (SNoof85) [inline source/_integrations/template.markdown]: number ?
Isn't that select ?

**Review by SNoof85 — COMMENTED** [review]
Numbers instead of select ?

File: source/_integrations/template.markdown [hunk 2]
```diff
@@ -168,16 +162,50 @@ binary_sensor:
...
+      description: Template for the select's available options.
+      required: true
+      type: template
+    optimistic:
+      description: Flag that defines if select works in optimistic mode.
+      required: false
+      type: boolean
+      default: false
+"[all sensor, binary sensor, select entities]":
+  description: Fields that can be used above for sensors, binary sensors, and numbers.
```
Comment (SNoof85) [inline source/_integrations/template.markdown:227]: number but no select

File: source/_integrations/template.markdown (hunk 1 above)
Comment (raman325) [reply source/_integrations/template.markdown]: thanks, copied from the other PR and forgot to replace all 😉

File: source/_integrations/template.markdown (hunk 2 above)
Comment (raman325) [reply source/_integrations/template.markdown:227]: thanks, copied from the other PR and forgot to replace all 😉

**Review by frenck — APPROVED** [review]

**Pull request merged**